datalake/src/main/java/com/sequenceiq/datalake/service/consumption/ConsumptionClientService.java

```diff
@@ -25,32 +25,32 @@ public ConsumptionClientService(ConsumptionInternalCrnClient consumptionInternal
         this.webApplicationExceptionMessageExtractor = webApplicationExceptionMessageExtractor;
     }
 
-    public void scheduleStorageConsumptionCollection(String accountId, StorageConsumptionRequest request) {
+    public void scheduleStorageConsumptionCollection(String accountId, StorageConsumptionRequest request, String userCrn) {
```

Comment:
you can use ThreadbasedUserCrnProvider.getUserCrn here too, instead of adding a new parameter

Reply:
Is it better to have multiple call to Thread function rather than generate one and pass it as parameter?